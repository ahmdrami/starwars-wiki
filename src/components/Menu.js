import React from 'react';
import arrowLeft from '../assets/arrow-left.svg';
import Spinner from './Spinner';
const Menu = ({ isFetching, menu, onClick }) => {
   const root = Array.isArray(menu);
   return (
      <aside>
         { isFetching ? (
             <Spinner />
         ) : !root ? (
            Object.keys(menu).map((link, i) => (
               <button className="button" key={i} onClick={() => onClick(link)}>
                  {link}
               </button>
            ))
         ) : (
            <div className="child-nav">
               <button className="button" onClick={() => onClick('back')}>
                  <img src={ arrowLeft } alt="Back button"/>
               </button>
               { menu.map((link, i) => (
                  <button
                     className="button"
                     key={i}
                     onClick={() => onClick(link)}
                  >
                     {link.name ? link.name : link.title}
                  </button>
               ))}

              
            </div>
         )}
      </aside>
   );
};

export default Menu;
