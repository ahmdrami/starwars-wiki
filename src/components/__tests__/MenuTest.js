import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import Menu from '../Menu';

configure({ adapter: new Adapter() });
const testProps = {
   isFetching: true,
   onClick: jest.fn(),
   menu: ['a', 'b']
};
describe('Menu', () => {
   it('Should be in a loading state when isFetching prop is true', () => {
      const wrapper = shallow(<Menu {...testProps} />);
      expect(wrapper.text()).toContain('Loading');
      expect(wrapper).toMatchSnapshot();

   });

   it('should display menu items when data is fetched and should not contain Loading text', () => {
      testProps.isFetching = false;
      const wrapper = shallow(<Menu {...testProps} />);
      expect(wrapper.find('button').length).toEqual(3); 
      expect(wrapper.text()).not.toContain('Loading');
   })
});
