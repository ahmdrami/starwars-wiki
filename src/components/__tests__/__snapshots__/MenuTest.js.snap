// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Should be in a loading state when isFetching prop is true 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Menu
    isFetching={true}
    menu={
        Array [
            "a",
            "b",
          ]
    }
    onClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Spinner />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "aside",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Spinner />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "aside",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
